sound/soc/sof/loader.c

```diff
@@ -47,7 +48,7 @@ int snd_sof_fw_parse_ext_data(struct snd_sof_dev *sdev, u32 bar, u32 offset)
 	/* get first header */
 	snd_sof_dsp_block_read(sdev, bar, offset, ext_data,
 			       sizeof(*ext_hdr));
-	ext_hdr = (struct sof_ipc_ext_data_hdr *)ext_data;
+	ext_hdr = ext_data;
```

Comment:
I would remove ext_data completely here, use ext_hdr everywhere, then also `(u8 *)ext_data + sizeof(*ext_hdr)` would become just `ext_hdr + 1`

Reply:
@lyakh i thought about this but I cant really do this can I? ext_hdr is just the header and can I allocate PAGE_SIZE to it?

Reply:
let's be careful, this code was updated to remove void pointer arithmetic recently. we can do optimizations in a follow-up patch, let's just remove spurious casts here.